jsonschema-use-cases.xml

```diff
@@ -0,0 +1,300 @@
+                "number", "string", "boolean", "true", "false", and "null" in this document are to
+                be interpreted as defined in <xref target="RFC8259">RFC 8259</xref>.
+            </t>
+
+            <t>
+                Use Cases describe the system's behavior under various conditions, cataloging who does what with the system, for what purpose, but without concern for system design or implementation.
+            </t>
+
+            <t>
+                Requirements list functional and non-functional or quality requirements, and the use cases they may be derived from/related to.
+            </t>
+        </section>
+
+        <section title="Use Cases">
+            <section title="Structural validation" anchor="uc-structural">
+                <t>
+                    The most basic function of validating JSON is the ability to express any subset of JSON that can be described by a deterministic context-free grammar, except for distinguishing between different encodings of the same value. Schemas that limit themselves to this functionality may be compiled to a deterministic pushdown automata.
+                </t>
+                <t>
+                    The following features of JSON are easily expressible as a pushdown automata:
+                </t>
+                <list>
+                    <t>JSON primitive type (string, object, etc)</t>
+                    <t>Minimum or maximum in a linear range of values</t>
+                    <t>Minimum or maximum lengths (number of characters, digits, items, or properties)</t>
+                    <t>Literal/constant values or alternate/enumerated values</t>
+                    <t>Pattern matching strings by a regular expression (including object keys)</t>
+                    <t>Logic operators (union, intersection, difference)</t>
+                    <t>Descent into object properties and array items</t>
+                </list>
+                <t>
+                    Multiple forms that are value-equal according JSON need not be distinguished under this use-case. This includes the ordering of properties in an object, character escapes in strings, and whitespace.
+                </t>
+            </section>
+
+            <section title="Semantic annotation" anchor="uc-annotations">
+                <t>
+                    There is a need to annotate values within a JSON document: for machine readability, and for documentation purposes.
+                    Given a document and directions for annotating it, you should should be able to:
+                </t>
+                <list>
+                    <t>document the meaning of a property,</t>
+                    <t>suggest a default value for new documents of a given type,</t>
+                    <t>fill in missing values (in objects or arrays) with values of equivalent meaning,</t>
+                    <t>generate a list of hyperlinks,</t>
```

Comment:
This seems very specific to hyperschema.

Reply:
That's exactly what it's discussing... but it makes sense here, the goal is to list features that "ought to be possible" and not necessarily "must be supported by everyone."